src/main/java/com/mixpanel/mixpanelapi/MixpanelAPI.java

```diff
@@ -200,5 +201,6 @@ private String slurp(InputStream in) throws IOException {
 
     private static final int BUFFER_SIZE = 256; // Small, we expect small responses.
     private static final int READ_TIMEOUT_MILLIS = 120000; // Two minutes should be more than enough for a response.
+    private static final int CONNECT_TIMEOUT_MILLIS = 30000; // Two minutes should be more than enough for a response.
```

Comment:
Sloppy of me. Fixed in 45c23720bfd148fa6e4912ab315d9b69a0fe5196. Thanks for feedback.